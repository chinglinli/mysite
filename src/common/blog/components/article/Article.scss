.article {
  background: white;
  padding: 32px;

  header {
    h1 {
      margin: 3px 0;
      font-size: 2rem;

      @media (max-width: 1200px) {
        font-size: 1.3rem;

        overflow : hidden;
        text-overflow : ellipsis;
        white-space : nowrap;
      }

      a {
        font-size: 2rem;
      }
    }
  }

  ul.metadata {
    font-size: 0.9rem;

    @media (max-width: 1200px) {
      font-size: 0.5rem;

      overflow : hidden;
      text-overflow : ellipsis;
      white-space : nowrap;
    }

    li {
      display: inline;
      &:not(:first-child) {
        margin-left: 8px;
      }
    }

    ul.categories {
      display: inline;

      li {
        margin: 0;
        &:not(:first-child):before {
          content: "/";
          margin: 0 3px;
        }
      }
    }
  }
}