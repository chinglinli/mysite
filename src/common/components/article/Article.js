import React from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import { Link } from 'react-router'

import Loading from '../loading'
import SiteHead from '../site-head'
import ArticleMeta from '../article-meta'
import ArticleContent from '../article-content'
import SocialShare from '../social-share'
import ArticleComment from '../article-comment'

import styles from './Article.scss'

export default class Article extends React.PureComponent {
  static PropTypes = {
    siteConfig: ImmutablePropTypes.map.isRequired,
    article: ImmutablePropTypes.contains({
      slug: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      categories: ImmutablePropTypes.list.isRequired,
      content: PropTypes.any.isRequired,
      rawSummary: PropTypes.string.isRequired
    }),
    socialConfig: ImmutablePropTypes.map.isRequired,
    commentConfig: ImmutablePropTypes.map.isRequired
  }

  render () {
    const {article, socialConfig, commentConfig} = this.props
    if (!article) {
      return (
        <article className={styles.root}>
          <Loading />
        </article>
      )
    }

    return (
      <article className={styles.root}>
        <SiteHead config={this.props.siteConfig}/>
        <div>
          <header className={styles.header}>
            <Link className={styles.link} to={`/articles/${article.get('slug')}/`}>{article.get('title')}</Link>
          </header>
          <ArticleMeta
            categories={article.get('categories')}
            date={article.get('date')}
            modifiedDate={article.get('modifiedDate')}
          />
        </div>
        <ArticleContent content={article.get('content')}/>
        <SocialShare config={socialConfig}/>
        <ArticleComment config={commentConfig}/>
      </article>
    )
  }
}

