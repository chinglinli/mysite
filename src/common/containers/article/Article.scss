@import '../../css/base.scss';

.root {
  padding: 2em;
  border-bottom: 1px solid #eee;
  background: white;

  .header {
    font-size: 2em;
    font-weight: bold;
    color: $dominant-color;

    .link {
      @extend .link;
    }
  }
}
