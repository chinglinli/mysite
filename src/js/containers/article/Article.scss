@import '../../../css/base.scss';

.root {
    padding: 2em;
    border-bottom:1px solid #eee;
    background: white;

    .header {
        font-size: 2em;
        color: $dominant-color;

        .link {
            @extend .link;
        }
    }
}
