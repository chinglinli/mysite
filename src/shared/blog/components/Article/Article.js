import React from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import Link from '../../../generic/components/Link'
import dateformat from 'dateformat'

import Loading from '../Loading'
import SocialShare from './SocialShare'
import Metadata from './Metadata'
import ArticleComment from './ArticleComment'
import ArticleContent from '../ArticleContent'

import styles from './Article.scss'

export default class Article extends React.PureComponent {
  static PropTypes = {
    summaryMode: PropTypes.bool,
    article: ImmutablePropTypes.contains({
      slug: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      categories: ImmutablePropTypes.listOf(ImmutablePropTypes.contains({
        slug: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      })).isRequired,
      date: PropTypes.instanceOf(Date).isRequired,
      modifiedDate: PropTypes.instanceOf(Date),
      content: PropTypes.any.isRequired,
      summary: PropTypes.string.isRequired
    }),
    socialConfig: ImmutablePropTypes.map,
    commentConfig: ImmutablePropTypes.map
  }

  renderMetadata = () => {
    const {article} = this.props
    return (
      <Metadata
        categories={article.get('categories')}
        date={article.get('date')}
        modifiedDate={article.get('modifiedDate')} />
    )
  }

  renderSummary = () => {
    return <ArticleContent content={this.props.article.get('summary')}/>
  }

  renderContent = () => {
    const {article, socialConfig, commentConfig} = this.props
    return [
      <ArticleContent key="article-content" content={article.get('content')}/>,
      <SocialShare  key="social-share" config={socialConfig}/>,
      <ArticleComment key="comment" config={commentConfig}/>
    ]
  }

  render () {
    const {article, summaryMode} = this.props
    if (!article) {
      return (
        <article className={styles.article}>
          <Loading/>
        </article>
      )
    }

    return (
      <article className={styles.article}>
        <header>
          <h1><Link to={`/articles/${article.get('slug')}/`}>{article.get('title')}</Link></h1>
          {this.renderMetadata()}
        </header>
        {summaryMode ? this.renderSummary() : this.renderContent()}
      </article>
    )
  }
}

