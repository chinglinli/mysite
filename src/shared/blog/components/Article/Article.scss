.article {
  background: white;
  padding: 32px;

  header {
    h1 {
      margin: 3px 0;
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;

      a {
        font-size: 2rem;
      }
    }
  }
}