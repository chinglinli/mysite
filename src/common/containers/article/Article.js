import * as React from 'react'
import ImmutablePropTypes from 'react-immutable-proptypes'
import {connect} from 'react-redux'
import {Link} from 'react-router'

import ArticleMeta from '../../components/article-meta'
import ArticleContent from '../../components/article-content'
import * as articleActions from '../../ducks/article/actions'
import * as articleSelectors from '../../ducks/article/selectors'

import styles from './Article.scss'


export class Article extends React.Component {
  static PropTypes = {
    article: ImmutablePropTypes.contains({
      slug: React.PropTypes.string.isRequired,
      title: React.PropTypes.string.isRequired,
      content: React.PropTypes.content
    })
  }

  componentWillMount() {
    if (!this.props.article) {
      this.props.fetchArticles()
    }
  }

  render() {
    const {article} = this.props
    if (!article) {
      return <article>讀取中……</article>
    }

    return (
      <article className={styles.root}>
        <div>
          <header className={styles.header}>
            <Link className={styles.link} to={`/articles/${article.get('slug')}/`}>{article.get('title')}</Link>
          </header>
          <ArticleMeta
            categories={article.get('categories')}
            date={article.get('date')}
            modifiedDate={article.get('modifiedDate')}
          />
        </div>
        <ArticleContent content={article.get('content')}/>
      </article>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    article: articleSelectors.getArticle(state, ownProps)
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    fetchArticles: () => dispatch(articleActions.fetchArticles())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Article)
