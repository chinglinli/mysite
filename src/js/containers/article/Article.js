import * as React from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { connect } from 'react-redux';

import Link from '../../components/link';
import ArticleMeta from '../../components/article-meta';
import * as articleActions from '../../ducks/article/actions';

import styles from './Article.scss';


export class Article extends React.Component {
    static PropTypes = {
        article: ImmutablePropTypes.contains({
            slug: React.PropTypes.string.isRequired,
            title: React.PropTypes.string.isRequired,
            content: React.PropTypes.content
        })
    };

    componentWillMount() {
        if(!this.props.article) {
            this.props.fetchArticles();
        }
    }

    render() {
        const { article } = this.props;
        return (
            <article className={styles.root}>
                <div>
                    <header className={styles.header}>
                        <Link to={`/articles/${article.get('slug')}/`}>{article.get('title')}</Link>
                    </header>
                    <ArticleMeta
                        categories={article.get('categories')}
                        date={article.get('date')}
                        modifiedDate={article.get('modified_date')}
                    />
                </div>
                <div dangerouslySetInnerHTML={{ __html: article.get('content')}} />
            </article>
        );
    }
}

const mapStateToProps = (state, ownProps) => {
    return {
        article: state.getIn(['article', 'items']).find(article => article.slug === ownProps.slug)
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        fetchArticles: () => dispatch(articleActions.fetchArticles())
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Article);
