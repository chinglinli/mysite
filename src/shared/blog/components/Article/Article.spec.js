import Immutable from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Article from './Article'

test('It should render Article component correctly', () => {
  const article = Immutable.fromJS({
      slug: 'slug',
      title: 'title',
      categories: [{
        slug: 'slug',
        name: 'name'
      }],
      date: new Date(2017, 6, 9),
      content: 'content',
      rawSummary: 'raw_summary'
  })
  const component = renderer.create(
    <Article article={article}/>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('It should render Article component correctly when the article is null', () => {
  const component = renderer.create(
    <Article article={null}/>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
