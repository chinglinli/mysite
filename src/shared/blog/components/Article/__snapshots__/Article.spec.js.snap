// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render Article component correctly 1`] = `
<article
  className="article"
>
  <header>
    <h1>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        title
      </a>
    </h1>
    <ul
      className="Metadata__ItemGroup-dlupfj-0 llqsil"
    >
      <li
        className="Metadata__Item-dlupfj-1 fpRYfi"
      >
        分類：
        <ul
          className="Metadata__Categories-dlupfj-2 CNRul"
        >
          <li
            className="Metadata__Category-dlupfj-3 cJIpQm"
          >
            <a
              className="NormalLink-s877q9z-0 coaBWT"
              onClick={[Function]}
              style={Object {}}
            >
              name
            </a>
          </li>
        </ul>
      </li>
      <li
        className="Metadata__Item-dlupfj-1 fpRYfi"
      >
        發表時間：
        2017/07/09
      </li>
    </ul>
  </header>
  <div
    className="content"
    dangerouslySetInnerHTML={
      Object {
        "__html": "content",
      }
    }
  />
  <div
    className="SocialShare__IconGroup-hghy08-0 gNCtfE"
  />
  <div />
</article>
`;

exports[`It should render Article component correctly when the article is null 1`] = `
<article
  className="article"
>
  <div
    className="Loading__Base-s1d02ruw-0 idafHP"
  >
    <div
      className="Loading__LoadingImage-s1d02ruw-1 cjrynz"
    />
    <div
      className="Loading__Message-s1d02ruw-2 fgoCqC"
    >
      正在努力讀取中…
    </div>
  </div>
</article>
`;
